### ISSUE TITLE
Problem with taup pierce for multiple diffractions

### ISSUE THREAD

**Issue by johnrudge** [crotwell/TauP#19]
There seems to be a bug in the times outputted by taup pierce with multiple diffractions. An example is
```
taup pierce -deg 210 -ph PdiffPdiff
```
```
PdiffPdiff at  1697.88 seconds at   210.00 degrees for a      0.0 km deep source in the iasp91 model with rayParam    4.439 s/deg.
    0.00     0.0     0.0
   -0.04    20.0     3.5
   -0.08    35.0     5.9
   -0.64   210.0    28.8
   -1.38   410.0    53.6
   -2.54   660.0    81.9
  -49.20  2889.0   409.8
  -55.80  2889.0   424.5
 -102.46   660.0   752.4
 -103.62   410.0   780.7
 -104.36   210.0   805.5
 -104.92    35.0   828.4
 -104.96    20.0   830.7
 -105.00     0.0   834.3
 -105.04    20.0   837.8
 -105.08    35.0   840.2
 -105.64   210.0   863.1
 -106.38   410.0   887.9
 -107.54   660.0   916.2
 -154.20  2889.0  1244.1
 -160.80  2889.0  1258.8
 -207.46   660.0  1586.7
 -208.62   410.0  1615.0
 -209.36   210.0  1639.8
 -209.92    35.0  1662.6
 -209.96    20.0  1665.0
 -210.00     0.0  1668.6
```
The discrepancy is between the first time quoted (1697.88 seconds) and the last time of the pierce points (1668.6 seconds). I think it arises in line 2944 of SeismicPhase.java.
```
double refractTime = (refractDist*currArrival.getRayParam()) / headOrDiffractSeq.size()
```
I don't think it should be `/ headOrDiffractSeq.size()` as that's already taken care of in `refractDist`

**Comment by johnrudge** [comment]
Related to this, I think there is also an issue with head waves missing the head segment. For example,
```
taup pierce -deg 1 -ph Pn
```
```
Pn at    21.27 seconds at     1.00 degrees for a      0.0 km deep source in
    0.00     0.0     0.0
    0.19    20.0     5.0
    0.37    35.0     8.9
    0.56    20.0    12.8
    0.74     0.0    17.7
```
the distance of return is 0.74 when it should be 1.0

**Comment by crotwell** [comment]
Believe both are fixed in 68523d87cc1f8056b1dbac7cf9d0018519f32171

Please verify, and thank you for the bug reports.

```
bin/taup pierce -deg 1 -ph Pn
> Pn at    21.27 seconds at     1.00 degrees for a      0.0 km deep source in the iasp91 model with rayParam   13.754 s/deg.
    0.00     0.0     0.0
    0.19    20.0     5.0
    0.37    35.0     8.9
    0.63    35.0    12.4
    0.81    20.0    16.3
    1.00     0.0    21.3
```

```
bin/taup pierce -deg 210 -ph PdiffPdiff
> PdiffPdiff at  1697.88 seconds at   210.00 degrees for a      0.0 km deep source in the iasp91 model with rayParam    4.439 s/deg.
    0.00     0.0     0.0
   -0.04    20.0     3.5
   -0.08    35.0     5.9
   -0.64   210.0    28.8
   -1.38   410.0    53.6
   -2.54   660.0    81.9
  -49.20  2889.0   409.8
  -55.80  2889.0   439.1
 -102.46   660.0   767.1
 -103.62   410.0   795.4
 -104.36   210.0   820.1
 -104.92    35.0   843.0
 -104.96    20.0   845.4
 -105.00     0.0   848.9
 -105.04    20.0   852.5
 -105.08    35.0   854.9
 -105.64   210.0   877.7
 -106.38   410.0   902.5
 -107.54   660.0   930.8
 -154.20  2889.0  1258.8
 -160.80  2889.0  1288.1
 -207.46   660.0  1616.0
 -208.62   410.0  1644.3
 -209.36   210.0  1669.1
 -209.92    35.0  1691.9
 -209.96    20.0  1694.3
 -210.00     0.0  1697.9

```
